plugins/cpp/parser/src/cppparser.cpp

```diff
@@ -329,19 +332,137 @@ int CppParser::worker(const clang::tooling::CompileCommand& command_)
 
 CppParser::CppParser(ParserContext& ctx_) : AbstractParser(ctx_)
 {
-  (util::OdbTransaction(_ctx.db))([&, this] {
-    for (const model::BuildAction& ba : _ctx.db->query<model::BuildAction>())
-      _parsedCommandHashes.insert(util::fnvHash(ba.command));
-  });
 }
-  
+
 std::vector<std::string> CppParser::getDependentParsers() const
 {
   return std::vector<std::string>{};
 }
 
+void CppParser::preparse()
+{
+  std::vector<model::FilePtr> filePtrs(_ctx.fileStatus.size());
+
+  std::transform(_ctx.fileStatus.begin(),
+                 _ctx.fileStatus.end(),
+                 filePtrs.begin(),
+                 [this](const auto& item)
+                 {
+                   if (item.second == IncrementalStatus::MODIFIED)
+                   {
+                     return _ctx.srcMgr.getFile(item.first);
+                   }
+                   else
+                   {
+                     return std::make_shared<model::File>();
+                   }
+                 });
+
+  (util::OdbTransaction(_ctx.db))([&]
+  {
+    for(const model::FilePtr file : filePtrs)
+    {
+      if(file)
+      {
+        markAsModified(file);
+      }
+    }
+
+    for(auto& item : _ctx.fileStatus)
```

Comment:
Spaces missing around these parts, again.

Reply:
Solved in 4b52c300da8e3f8309356cf71e5f9c70207cc051